test/tools/cli/IonProcess.java

```diff
@@ -0,0 +1,312 @@
+            } else {
+                switch (defaultValue){
+                    case SYSTEM_OUT_DEFAULT_VALUE:
+                        outputStream = new BufferedOutputStream(System.out, BUFFER_SIZE);
+                        break;
+                    case SYSTEM_ERR_DEFAULT_VALUE:
+                        outputStream = new BufferedOutputStream(System.err, BUFFER_SIZE);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        } catch (IOException e ) {
+            System.err.println("Initialize Output Stream Error:" + e.getMessage());
+        }
+        return outputStream;
+    }
+
+    /**
+     * this function is only used for CONTAINER_START and SCALAR eventType since other types don't need initial data.
+     */
+    private static Event ionStreamToEvent(EventType eventType,
+                                          IonReader ionReader) throws IOException, IllegalStateException {
+        if (ionReader.getType() == null) throw new IllegalStateException("Can't convert ionReader null type to Event");
+
+        IonType ionType = ionReader.getType();
+        SymbolToken fieldName = ionReader.getFieldNameSymbol();
+        SymbolToken[] annotations = ionReader.getTypeAnnotationSymbols();
+
+        String valueText = null;
+        byte[] valueBinary = null;
+        if (eventType == EventType.SCALAR) {
+            //write Text
+            ByteArrayOutputStream textOut = new ByteArrayOutputStream();
+            IonWriter textWriter = IonTextWriterBuilder.standard().build(textOut);
+            textWriter.writeValue(ionReader);
+            textWriter.finish();
+            valueText = textOut.toString();
+            //write binary
+            ByteArrayOutputStream binaryOut = new ByteArrayOutputStream();
+            IonWriter binaryWriter = IonBinaryWriterBuilder.standard().build(binaryOut);
+            binaryWriter.writeValue(ionReader);
+            binaryWriter.finish();
+            valueBinary = binaryOut.toByteArray();
+        }
+
+        ImportDescriptor[] imports = null;
+        int depth = ionReader.getDepth();
+        return new Event(eventType, ionType, fieldName, annotations, valueText, valueBinary, imports, depth);
+    }
+
+    private static boolean isSameSymbolTable(SymbolTable x, SymbolTable y) {
+        if (x.isSystemTable() && y.isSystemTable()) {
+            return (x.getVersion() == y.getVersion());
+        } else if (x.isSharedTable() && y.isSharedTable()) {
+            return (x.getName() == y.getName() & (x.getVersion()==y.getVersion()));
+        } else if (x.isLocalTable() && y.isLocalTable()) {
+            SymbolTable[] xTable = x.getImportedTables();
+            SymbolTable[] yTable = y.getImportedTables();
+            //compare imports
+            if (xTable.length == yTable.length) {
+                for (int i=0; i < xTable.length; i++) {
```

Comment:
Please always put a space around an `=`, i.e. `i = 0`

Reply:
Changed, I'll keep these styles in mind.

Do we have any tools or IDE plugin that checks styles in case I miss them?

Reply:
We don't have a standard style configuration for IntelliJ right now, but you can update your settings individually to enforce these things I think.

Reply:
Got it Thanks!